app/javascript/packages/analytics/index.ts

```diff
@@ -17,14 +17,30 @@ interface NewRelicGlobals {
  */
 export async function trackEvent(event: string, payload?: object): Promise<void> {
   const endpoint = getConfigValue('analyticsEndpoint');
+
   if (!endpoint) {
     return;
   }
 
+  const eventJson = JSON.stringify({ event, payload });
+
+  // Favor making analytics requests using sendBeacon(), which can be prioritized
+  // appropriately by the browser and have a better chance of succeeding during page unload.
+  try {
+    const blob = new Blob([eventJson], { type: 'application/json' });
+    const wasQueued = navigator.sendBeacon(endpoint, blob);
+    if (wasQueued) {
+      // The browser has promised it will send our request for us -- we can stop now.
+      return;
+    }
+  } catch {}
+
+  // Fall back to fetch() if sendBeacon is not available or if the call to sendBeacon() did
+  // not result in the browser actually making the request.
   await fetch(endpoint, {
```

Comment:
I found a [polyfill for sendBeacon](https://github.com/miguelmota/Navigator.sendBeacon) but it looks archived. Did we consider polyfilling?

Or alternatively, could we shim it ourselves?
```
if (window,navigator.sendBeacon === undefined) {
  window.navigator.sendBeacon = function(...) { window.fetch(...) }
}

Reply:
I think the only place we'd call it would be in trackEvent, so I don't think a polyfill would benefit us that much.

We also have to fall back to the fetch implementation if the browser-native sendBeacon returns false (indicating that the browser, for some reason, decided not to send our request).

Reply:
In the spirit of "fire and forget" it might be OK if the API returns `false` and we just forget it, right?

Reply:
Yeah I have not actually managed to find any docs on a case where the browser might actually return false, I think they just reserve the right to do so--presumably in conjunction with some sort of "ask websites not to track me" kind of preference. In that case, it maybe makes sense to just go with @aduth's suggestion and ditch the fallback.